### ISSUE TITLE
[FEA] Add `tolist()` method for Series/columns

### ISSUE THREAD

**Issue by rjzamora** [rapidsai/cudf#2479]
**Is your feature request related to a problem? Please describe.**
In pandas, a series object can be converted into a python list using the `tolist()` method. This method is currently unavailable in cudf.  Although there are obvious workarounds, the method should be a simple addition.

**Describe the solution you'd like**
I would like `tolist()` to return a list representation of a cudf series/column, to match the behavior in pandas. In pandas, the behavior looks like this:

```python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame({'x': [2, 3, 4], 'y':['a', 'b', 'c']})

In [3]: df['x'].tolist()
Out[3]: [2, 3, 4]
```

Note that this is different from the `values` property, which returns a numpy array:

```python
In [4]: df['x'].values
Out[4]: array([2, 3, 4])
```

**Describe alternatives you've considered**
The output of `df['x'].tolist()` should be identical to `[v for v in df['x']]`, and *usually* equvalent to `df['x'].values.tolist()`.

**Comment by rjzamora** [comment]
@brandon-b-miller - This might be a reasonable issue for you to get your feet wet (if you have free time)

**Comment by brandon-b-miller** [comment]
I am going to take a look at this.

**Comment by brandon-b-miller** [comment]
implemented in #2559
